src/modules/metrics/actions.js

```diff
@@ -22,11 +21,23 @@ export function addMetric(item) {
   }
 }
 
-//spaceId must be done before the metric is removed here.
 export function removeMetrics(ids) {
   return (dispatch, getState) => {
     if (ids.length === 0) { return }
+
     const spaceId = findSpaceId(getState, ids[0])
+
+    const {organizations, spaces, facts: {organizationFacts}} = getState()
+    const organizationId = e.collections.gget(spaces, spaceId, 'id', 'organization_id')
+    if (!!organizationId) {
+      const organization = e.collections.get(organizations, organizationId)
+      const facts = e.organization.findFacts(organizationId, organizationFacts)
+      const factsToDelete = _.filter(facts, f => f.exported_from_id === spaceId && ids.includes(f.metric_id))
+      factsToDelete.forEach(fact => {
```

Comment:
It would be nicer if they could be deleted as part of one action, though i realize that practically speaking one metric removal should only go to one fact removal.